contrib/benchmarkDictBuilder/benchmark.c

```diff
@@ -0,0 +1,458 @@
+        if (**stringPtr=='i') (*stringPtr)++;
+        if (**stringPtr=='B') (*stringPtr)++;
+    }
+    return result;
+}
+
+/** longCommandWArg() :
+ *  check if *stringPtr is the same as longCommand.
+ *  If yes, @return 1 and advances *stringPtr to the position which immediately follows longCommand.
+ * @return 0 and doesn't modify *stringPtr otherwise.
+ */
+static unsigned longCommandWArg(const char** stringPtr, const char* longCommand){
+    size_t const comSize = strlen(longCommand);
+    int const result = !strncmp(*stringPtr, longCommand, comSize);
+    if (result) *stringPtr += comSize;
+    return result;
+}
+
+static void fillNoise(void* buffer, size_t length)
+{
+    unsigned const prime1 = 2654435761U;
+    unsigned const prime2 = 2246822519U;
+    unsigned acc = prime1;
+    size_t p=0;;
+
+    for (p=0; p<length; p++) {
+        acc *= prime2;
+        ((unsigned char*)buffer)[p] = (unsigned char)(acc >> 21);
+    }
+}
+
+/*-*************************************
+* Dictionary related operations
+***************************************/
+/** createDictFromFiles() :
+ *  Based on type of param given, train dictionary using the corresponding algorithm
+ *  @return dictInfo containing dictionary buffer and dictionary size
+ */
+dictInfo* createDictFromFiles(sampleInfo *info, unsigned maxDictSize,
+                  ZDICT_random_params_t *randomParams, ZDICT_cover_params_t *coverParams,
+                  ZDICT_legacy_params_t *legacyParams) {
+    unsigned const displayLevel = randomParams ? randomParams->zParams.notificationLevel :
+                        coverParams ? coverParams->zParams.notificationLevel :
+                        legacyParams ? legacyParams->zParams.notificationLevel :
+                        0;   /* should never happen */
+    void* const dictBuffer = malloc(maxDictSize);
+
+    dictInfo* dInfo;
+
+    /* Checks */
+    if (!dictBuffer)
+        EXM_THROW(12, "not enough memory for trainFromFiles");   /* should not happen */
+
+    {   size_t dictSize;
+        if(randomParams) {
+          dictSize = ZDICT_trainFromBuffer_random(dictBuffer, maxDictSize, info->srcBuffer,
+                                               info->samplesSizes, info->nbSamples, *randomParams);
+        }else if(coverParams) {
+          dictSize = ZDICT_optimizeTrainFromBuffer_cover(dictBuffer, maxDictSize, info->srcBuffer,
+                                                info->samplesSizes, info->nbSamples, coverParams);
+        } else {
+          size_t totalSize= 0;
+          for (int i = 0; i < info->nbSamples; i++) {
+            totalSize += info->samplesSizes[i];
+          }
+          size_t const maxMem = findMaxMem(totalSize * MEMMULT) / MEMMULT;
+          size_t loadedSize = (size_t) MIN ((unsigned long long)maxMem, totalSize);
+          fillNoise((char*)(info->srcBuffer) + loadedSize, NOISELENGTH);
+          dictSize = ZDICT_trainFromBuffer_unsafe_legacy(dictBuffer, maxDictSize, info->srcBuffer,
```

Comment:
You can avoid the extra complexity by just using `ZDICT_trainFromBuffer_legacy()`.